modules/workflow-workflowoperation/src/main/java/org/opencastproject/workflow/handler/workflow/PublicationChannelToWorkspace.java

```diff
@@ -0,0 +1,153 @@
+ */
+@Component(
+    immediate = true,
+    service = WorkflowOperationHandler.class,
+    property = {
+        "service.description=move publication elements to workspace",
+        "workflow.operation=publication-channel-to-workspace"
+    }
+)
+
+public class PublicationChannelToWorkspace extends AbstractWorkflowOperationHandler {
+
+  /** Configuration key for the "source-channel" to use as a source input */
+  static final String OPT_SOURCE_PUBLICATION_CHANNEL = "source-channel";
+
+  /** The logging facility */
+  private static final Logger logger = LoggerFactory
+          .getLogger(PublicationChannelToWorkspace.class);
+
+  @Override
+  public WorkflowOperationResult start(WorkflowInstance workflowInstance, JobContext context)
+      throws WorkflowOperationException {
+
+    logger.info("Running get Publicationchannel to workspace for medipackage {}", workflowInstance.getId());
+    final MediaPackage mediaPackage = workflowInstance.getMediaPackage();
+
+    WorkflowOperationInstance currentOperation = workflowInstance.getCurrentOperation();
+
+    ConfiguredTagsAndFlavors tagsAndFlavors = getTagsAndFlavors(workflowInstance, Configuration.many,
+        Configuration.many, Configuration.many, Configuration.many);
+    List<MediaPackageElementFlavor> configuredSourceFlavors = tagsAndFlavors.getSrcFlavors();
+    List<String> configuredSourceTags = tagsAndFlavors.getSrcTags();
+    List <String> configuredTargetTags = tagsAndFlavors.getTargetTags();
+    String publicationChannel = StringUtils
+        .trimToEmpty(currentOperation.getConfiguration(OPT_SOURCE_PUBLICATION_CHANNEL));
+    String configuredTargetTagsAsString =  String.join(",", configuredTargetTags);
+
+    if (publicationChannel.isEmpty()) {
+      logger.error("No source publication-channel set. Operation will be skipped.");
+      return createResult(mediaPackage, Action.SKIP);
+    }
+    if (configuredSourceFlavors.isEmpty()) {
+      logger.error("No source flavor set. Operation will be skipped.");
+      return createResult(mediaPackage, Action.SKIP);
+    }
+    if (configuredSourceTags.isEmpty()) {
+      logger.error("No source Tag set. Operation will be skipped.");
+      return createResult(mediaPackage, Action.SKIP);
+    }
+
+    Optional<Publication> publication = Arrays.stream(mediaPackage.getPublications())
+        .filter(channel -> channel.getChannel().equals(publicationChannel)).findFirst();
+    if (publication.get().getTracks().length >= 0) {
+
+      Collection<MediaPackageElement> tracks =  new ArrayList<MediaPackageElement>();
+
+      Arrays.stream(publication.get().getTracks())
+          .filter(element -> element.containsTag(configuredSourceTags))
+          .filter(Objects::nonNull)
+          .forEach(element -> tracks.add(element));
+      Arrays.stream(publication.get().getTracks())
+          .filter(element -> configuredSourceFlavors.contains(element.getFlavor()))
+          .filter(Objects::nonNull)
```

Comment:
same here
```suggestion
```

Reply:
i did this, because if there is no Flavor i had nullpointer exceptions.
So what is your suggestion here, why schould that be removed.
Maybe i misunderstood this stream filter?